genie-web/src/main/java/com/netflix/genie/web/agent/resources/AgentFileProtocolResolver.java

```diff
@@ -64,14 +66,19 @@ public AgentFileProtocolResolver(
      * @throws URISyntaxException if constructing the URI fails
      */
     public static URI createUri(final String jobId, final String path) throws URISyntaxException {
-        return new URI(AgentFileProtocolResolver.URI_SCHEME, jobId, path, null);
+        final String encodedJobId = Base64.encodeBase64URLSafeString(jobId.getBytes());
```

Comment:
Now that you're encoding the ID are you missing decoding it somewhere?

I don't see it anywhere